internal/pkg/k8s/client.go

```diff
@@ -23,35 +23,58 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/clientcmd/api"
+	"os"
+	"sync"
 )
 
-func init() {
-	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
-	// use the current context in kubeconfig
-	clientConfig, err := kubeconfig.ClientConfig()
-	if err != nil {
-		panic(err.Error())
-	}
+var (
+	cachedClient *kubernetes.Clientset
+	cachedConfig *api.Config
 
-	Config, err = loadingRules.Load()
-	if err != nil {
-		panic(err.Error())
-	}
+	once sync.Once
+)
+
+func initClient() {
+	err := func() error {
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+		// use the current context in kubeconfig
+		clientConfig, err := kubeconfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+
+		cachedConfig, err = loadingRules.Load()
+		if err != nil {
+			return err
+		}
+
+		cachedClient, err = kubernetes.NewForConfig(clientConfig)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
 
-	Client, err = kubernetes.NewForConfig(clientConfig)
 	if err != nil {
-		panic(err.Error())
+		fmt.Fprintf(os.Stderr, "failed to configure Kubernetes cluster context: %s\n", err)
```

Comment:
I would file this as a TODO, but prefer to use klog/glog for logging that adds timestamps, filtering on V level etc.